ArduCopter/mode.cpp

```diff
@@ -194,6 +194,133 @@ void Copter::mode_change_failed(const Mode *mode, const char *reason)
     }
 }
 
+
+// Check if this mode can be entered from the GCS
+bool Copter::gcs_mode_enabled(Mode::Number mode_num)
+{
+    // We don't have parameter space to use the mode numbers in the bitmask directly, convert to reduced range
+    uint8_t mode_bit = 0;
+    switch (mode_num) {
+        case Mode::Number::RTL:
+        case Mode::Number::LAND:
+            // These modes are always enabled
+            return true;
+
+        case Mode::Number::STABILIZE:
```

Comment:
a list exposed in AP_Vehicle, where index into list is the bit would be most compact, and would allow for common code